torch/csrc/autograd/profiler.h

```diff
@@ -71,59 +153,58 @@ struct RangeEventList {
 };
 
 extern bool profiling;
+extern bool using_nvprof;
```

Comment:
Can you add a comment saying that `using_nvprof == true` doesn't imply `using_cuda == true` and that they are two distinct modes? Maybe we should just use an enum for profiler state?
```cpp
class enum ProfilerState {
  disabled,
  CPU,          // CPU-only profiling
  CUDA,         // CPU + CUDA events
  NVTX          // only emit NVTX markers
};
```